src/common/snippets/src/lowered/pass/serialize_control_flow.cpp

```diff
@@ -9,16 +9,22 @@
 #include "snippets/lowered/linear_ir.hpp"
 #include "snippets/op/serialization_node.hpp"
 #include "snippets/snippets_isa.hpp"
+#include "snippets/lowered/loop_manager.hpp"
+#include "snippets/lowered/linear_ir_builder.hpp"
 
 namespace ov {
 namespace snippets {
 namespace lowered {
 namespace pass {
 
-bool SerializeControlFlow::run(LinearIR& linear_ir) {
+bool SerializeControlFlow::run(const LinearIR& original_linear_ir) {
     OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::SerializeControlFlow")
-    if (linear_ir.empty())
+    if (original_linear_ir.empty())
         return false;
+    const auto& linear_ir = m_update_dynamic_ops ? LinearIRBuilder().clone(original_linear_ir) : original_linear_ir;
```

Comment:
Maybe we can always update dynamic ops and remove `m_update_dynamic_ops` field? If we left this field, we would have to handle it in debug caps (e.g. in upcoming PR #24293). It looks like unnecessary complication.
Yes, in this case we will always have to clone the LIR, but is that a problem? From my opinion, performance overheads are not so important in this case.

Reply:
From my side: I'm not sure that we should always serialize updated dynamic ops because sometimes I want to see real LinearIR with dynamic parameters. So I'd prefer to have flag to turn on/off the needed representation.
Another idea is to create another pass `SerializeRuntimeControlFlow` where we make a copy of linear IR, update dynamic ops and call `SerializeControlFlow` pass with updated cloned LinearIR to avoid this flag

Reply:
@IvanNovoselov @a-sidorova got your points, thank you. Let's leave the current behavior